### PR TITLE
fix(semantic-release): Update semantic-release to latest

### PR DESCRIPTION
**Pull request by arajkumar**
* Updated semantic-release to latest
* Added package-lock.json

### CODE REVIEW

**Comment by arajkumar** [comment]
[test]

**Comment by arajkumar** [comment]
[test]

**Comment by arajkumar** [comment]
It is still WIP. Don't merge/approve.

File: package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
     "chai": "^4.2.0",
     "mocha": "^6.2.0",
     "nyc": "^14.1.1",
-    "semantic-release": "8.2.0",
```
Comment (invincibleJai) [inline package.json]: am n ot sure if it'll help , there was some issue with 15.x in CI , chek this https://github.com/fabric8-analytics/fabric8-analytics-lsp-server/pull/85

Comment (arajkumar) [reply package.json]: Thanks @invincibleJai . I had updated the build scripts to work with latest semantic-release module. PTAL.

**Comment by arajkumar** [comment]
@invincibleJai 

I did used my personal npmjs account and forked github repo to test this change, something like below,
```diff
diff --git a/package.json b/package.json
index 179cbf2..6ee379b 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "ararunrajkumar-lsp-server-demo",
+  "name": "fabric8-analytics-lsp-server",
   "description": "LSP Server for Dependency Analytics",
   "version": "0.0.0-development",
   "author": "Pavel Odvody",
@@ -27,7 +27,7 @@
   ],
   "repository": {
     "type": "git",
-    "url": "https://github.com/arajkumar/fabric8-analytics-lsp-server.git"
+    "url": "https://github.com/fabric8-analytics/fabric8-analytics-lsp-server.git"
   },
   "dependencies": {
     "request": "^2.79.0",
```

File: package.json [hunk 2]
```diff
@@ -75,8 +77,17 @@
   "release": {
     "branch": "master",
     "debug": false,
-    "verifyConditions": {
-      "path": "./node_modules/@krux/condition-jenkins"
-    }
+    "prepare": [
```
Comment (invincibleJai) [inline package.json]: is `verifyConditions` no more needed?

**Comment by invincibleJai** [comment]
@arajkumar looks like CI failed for this change, let me retrigger it https://ci.centos.org/job/devtools-fabric8-analytics-lsp-server/88/

**Comment by invincibleJai** [comment]
[test]

**Comment by arajkumar** [comment]
> @arajkumar looks like CI failed for this change, let me retrigger it https://ci.centos.org/job/devtools-fabric8-analytics-lsp-server/88/

There is a typo in the script. I will fix it ASAP. Thanks!

File: package.json (hunk 2 above)
Comment (arajkumar) [reply package.json]: It used to check the existence of env variables to find whether we running from CI env or not. Now it is not required.

File: package.json [hunk 3]
```diff
@@ -37,23 +37,25 @@
...
   },
   "scripts": {
-    "build": "npm run clean && node node_modules/typescript/bin/tsc -p . && cp LICENSE package.json README.md output && npm run dist",
+    "build-semantic": "npm run clean && node node_modules/typescript/bin/tsc -p . && cp LICENSE package.json README.md output",
+    "build": "npm run build-semantic && npm run dist",
     "clean": "rm -Rf ca-lsp-server.tar output/",
     "test": "nyc mocha",
-    "dist": "cp -r node_modules output/ && cp ./package.json output/ && node -p -e \"require('./package.json').version\" > output/VERSION && rm -rf output/node_modules/typescript/ && tar cvjf ca-lsp-server.tar -C output/ .",
-    "semantic-release": "semantic-release pre && npm run build && cp -r .git output && npm publish output/ && semantic-release post"
+    "semantic-pre": "cp -a node_modules output/ && cp ./package.json output/",
```
Comment (invincibleJai) [inline package.json]: pls validate `ca-lsp-server.tar` locally with the one in release once, it's being used by other clients. It's good you tried to archive `node_moodules`

**Comment by arajkumar** [comment]
[test]

**Comment by arajkumar** [comment]
[test]

**Comment by arajkumar** [comment]
[test]

**Comment by arajkumar** [comment]
@invincibleJai Looks like epel-release package is not required for this component, similar like vscode extension. Now the build is passing in CI.

**Comment by arajkumar** [comment]
[test]

**Comment by dvandra** [comment]
@arajkumar can we go ahead and merge this PR??

**Review by dvandra — APPROVED** [review]
LGTM

**Pull request merged**